platform/sponge/src/main/java/me/lokka30/treasury/plugin/sponge/TreasuryCommand.java

```diff
@@ -0,0 +1,160 @@
+                .shortDescription(Component.text("/treasury economy migrate <plugin1> <plugin2>"))
+                .addParameters(
+                        plugin1Param = Parameter
+                                .string()
+                                .key("plugin1")
+                                .completer((context, input) -> this.complete(
+                                        context,
+                                        new String[] {"economy", "migrate", input}
+                                )).build(),
+                        plugin2Param = Parameter
+                                .string()
+                                .key("plugin2")
+                                .completer((context, input) -> this.complete(
+                                        context,
+                                        new String[] {"economy", "migrate", input}
+                                ))
+                                .build()
+                )
+                .executor(ctx -> {
+                    Optional<String> firstPlugin = ctx.one(plugin1Param);
+                    Optional<String> secondPlugin = ctx.one(plugin2Param);
+                    if (!firstPlugin.isPresent() || !secondPlugin.isPresent()) {
+                        // sends usage
+                        return this.execute(ctx, new String[] {"economy", "migrate"});
+                    }
+                    return this.execute(
+                            ctx,
+                            new String[] {"economy", "migrate", firstPlugin.get(), secondPlugin.get()}
+                    );
+                })
+                .terminal(true)
+                .build();
+        //@formatter:on
+
+        Command.Builder builder = Command
+                .builder()
+                .permission("treasury.command.treasury.economy")
+                .shortDescription(Component.text("/treasury economy <sub> [args]"))
+                .terminal(true);
+        builder.addChild(migrate, "migrate");
+        try {
+            for (Map.Entry<String, Command.Parameterized> e : temp.entrySet()) {
+                builder.addChild(e.getValue(), e.getKey());
+            }
+        } finally {
+            temp.clear();
+        }
+
+        return builder.build();
+    }
+
+    private CommandResult execute(CommandContext context, String[] args) {
+        this.base.execute(this.sources.obtainSource(context.cause().audience()), "treasury", args);
+        return CommandResult.success();
+    }
+
+    private List<CommandCompletion> complete(CommandContext context, String[] args) {
+        List<String> completions = this.base.complete(this.sources.obtainSource(context
+                .cause()
+                .audience()), "treasury", args);
+
+        if (completions == null || completions.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        List<CommandCompletion> ret = new ArrayList<>();
```

Comment:
Set an initial size for this list
```suggestion
        List<CommandCompletion> ret = new ArrayList<>(completions.size());
```